pkg/operator/csidriveroperator/csioperatorclient/shared-resource.go

```diff
@@ -29,6 +29,9 @@ func GetSharedResourceCSIOperatorConfig() CSIOperatorConfig {
 			"csidriveroperators/shared-resource/06_clusterrolebinding.yaml",
 			"csidriveroperators/shared-resource/07_role_config.yaml",
 			"csidriveroperators/shared-resource/08_rolebinding_config.yaml",
+			"csidriveroperators/shared-resource/11_metrics_service.yaml",
+			"csidriveroperators/shared-resource/12_servicemonitor.yaml",
```

Comment:
`StaticAssets` can't handle `ServiceMontior`:

> Operator degraded (CSIDriverStarter_SyncError): CSIDriverStarterDegraded: no kind "ServiceMonitor" is registered for version "monitoring.coreos.com/v1" in scheme "pkg/runtime/scheme.go:100" | Operator degraded (CSIDriverStarter_SyncError): CSIDriverStarterDegraded: no kind "ServiceMonitor" is registered for version "monitoring.coreos.com/v1" in scheme "pkg/runtime/scheme.go:100"

Currently, we don't create ServiceMonitors for the driver operators in CSO. Maybe we should... [CSO already installs ServiceMonitor for vsphere-problem-detector](https://github.com/openshift/cluster-storage-operator/blob/master/pkg/operator/vsphereproblemdetector/monitoring.go), you could generalize that (move to a better place + replace few vSphere references?)

Or just don't install the ServiceMonitor and file a BZ to add it to all CSI driver operators, I think the fault is on our side.

Reply:
Hmm, or just update library-go? It could handle ServiceMonitors now, https://github.com/openshift/library-go/blob/7aab941b4a2b60dc5687744944789456d7a4eb4e/pkg/operator/resource/resourceapply/monitoring.go

Reply:
Just pushed a change that updates library-go, hopefully that will solve the issue.